### DISCUSSION TITLE
Why does convert-im6 always remove searchable text from a PDF file?

### DISCUSSION THREAD

**Discussion by wbrells** [ImageMagick/ImageMagick#4259] [Help] [answered]
The **convert-im6** routine does a wonderful job of reducing the size of PDF files, but no matter what options I use (or don't use) the compressed PDF file is no longer searchable! This "side effect" makes the compressed file just about useless, so I'm hoping someone can explain why this is happening and, hopeful, suggest some option that will preserve the PDF text in a way that allows searching. Thanks, Wayne

**Comment by snibgo** [comment, marked as answer]
IM is a raster image processor. It rasterizes the PDF. That is, it creates images (one image per PDF page) that are made up of pixels. The result contains no code from the PDF, no searchable text.

I suppose you could convert the PDF to a text format (using some other tool), and insert that as metadata into the raster image.

**Comment by fmw42** [comment]
ImageMagick is a raster processor. PDFs are vector files. When Imagemage reads a vector PDF, it rasterizes it into pixels as an image. When it writes the PDF, it simply writes the raster image into a vector PDF shell. So you loose all vector information such as text that could be searched.

I believe you can install potrace on Windows and get ImageMagick to use it to vectorize your file. But I have no experience with that and cannot say how well that works. One of the Windows users will have to comment further.

**Reply by wbrells** [reply]
Thanks for the detailed answer! Per my reply above, it appears I'll need find some other way of reducing my PDF file sizes!

**Comment by Jmuccigr** [comment]
Your success in reducing the size of your PDFs will depend on what's in them. If you're on a Mac, have a look at [PDF Toolkit+](http://molsim.info/mobsci/pdftoolkit.html).

**Comment by snibgo** [comment]
I'll mention that many specialist PDF tools such as "qpdf" are available on many platforms, including Windows (in Cygwin distributions). I don't know whether those can help you reduce PDF file sizes.

**Reply by wbrells** [reply]
snibgo, Thanks for the suggestion. I'm currently testing various Ghostscript options to see what kind of PDF file size reduction I can get. Using the -dPDFSETTINGS=/screen option cuts the file size about in half. The image quality isn't great but probably acceptable. It seems that Adobe Acrobat on Windows (using the Reduce File Size option) can do considerable better than Ghostscript. I was hoping there was some set of Ghostscript options that could approximate the Acrobat results, but so far I haven't been successful in doing so!  Wayne
